src/braket/parametric/free_parameter.py

```diff
@@ -99,6 +105,22 @@ def to_dict(self) -> dict:
             "name": self.name,
         }
 
+    @staticmethod
+    def _validate_name(name: str) -> None:
+        if not name:
+            raise ValueError("Symbol names must be non empty")
+        if not isinstance(name, str):
+            raise TypeError("Symbol name must be a string")
+        if name.startswith("_"):
```

Comment:
What is the purpose of the nested `if` statement?

Reply:
Names with one underscore are allowed. Names with two are not.

Reply:
This is kind of a big deal because it potentially breaks customers who were using two underlines or other unicode characters before. However, they weren't able to convert those to OpenQASM anyways. So I'm wondering if there's a use case where we let customers use those as long as they bind them afterwards? Or would it be too complicated to check when generating OpenQASM...

Reply:
This has been reduced as well